content/blog/bare-metal-provisioning-on-hpe-greenlake-using-terraform.md

```diff
@@ -0,0 +1,353 @@
+### Querying for available OS images
+
+In order to list the available OS images for OS, add the below data statements in your Terraform file ***main.tf***:
+
+```hcl
+data "hpegl_metal_available_images" "ubuntu" { 
+  # select anything that looks like ubuntu:20.04
+  filter {
+    name   = "flavor"
+    values = ["(?i)ubuntu"] 
+  }
+
+  filter {
+    name   = "version"
+    values = ["20.04*"] 
+  }
+}
+```
+
+ The OS image list can be fetched using the following statements in main.tf: 
+
+```hcl
+locals {
+  ubuntu_image = format("%s@%s", data.hpegl_metal_available_images.ubuntu.images[0].flavor,
+                          data.hpegl_metal_available_images.ubuntu.images[0].version)
+}
+```
+
+### Querying for other available resources
+
+For this, you should use **hpegl_metal_available_resources** data resource. For example, the following statements show how to retrieve and store the available SSH keys in a local variable.  
+
+Append to main.tf:
+
+```hcl
+# query available resources.
+data "hpegl_metal_available_resources" "available" {
+}
+
+# using one of the available SSH keys.
+locals  {
+  ssh_keys = data.hpegl_metal_available_resources.available.ssh_keys[0].name
+}
+```
+
+Using a similar technique, you can retrieve the rest of the data you need - networks, machine size, etc.  
+
+A﻿ppend to main.tf:
+
+```hcl
+# choosing a location that has at least one machine available.
+locals {
+  location = ([for msize in data.hpegl_metal_available_resources.available.machine_sizes : msize.location 
+                    if msize.quantity > 0])[0]
+}
+
+# Listing required networks for this location.
+locals {
+  networks = ([for net in data.hpegl_metal_available_resources.available.networks : net.name 
+                  if net.host_use == "Required"  && net.location == local.location])
+}
+
+# choosing machine size/Compute Instance Type to deploy OS.
+locals {
+  machine_size = ([for msize in data.hpegl_metal_available_resources.available.machine_sizes : msize.name 
+                    if msize.location == local.location])
+}
+```
+
+[Here](https://registry.terraform.io/providers/HPE/hpegl/latest/docs/data-sources/metal_available_resources) you can get information about each of the bare metal data statements supported by the **hpegl** provider.
+
+### Create Compute instance
+
+The last step is for you define a **hpegl_metal_host** terraform resource in the file to request a new compute instance (host):
+
+```hcl
+resource "hpegl_metal_host" "demo_host" {
+  name          = "demo-host-1"
+  image         = local.ubuntu_image
+  machine_size  = local.machine_size[0]
+  ssh           = [ local.ssh_keys ]
+  networks      = local.networks
+  location      = local.location
+  description   = "Simple Host Deployment Demo"
+}
+```
+
+#### Initialize Terraform
+
+Before you can use Terraform, you will have to initialize it from the configuration file we have created. In the same directory as the **main.tf** file you created, **run : terraform init.**
```

Comment:
Suggestion:
In the same directory as the **main.tf** file you created, run the command: **terraform init**.